src/store/transactions/fungible-token-transfer.ts

```diff
@@ -34,4 +50,66 @@ export const makeFungibleTokenTransferState = atom(get => {
   return;
 });
 
+interface AssetTransferTxOptions {
+  amount: number;
+  balances: any;
+  memo: string;
+  network: StacksNetwork;
+  nonce: number;
+  recipient: string;
+  selectedAsset: AssetWithMeta;
+  senderKey: string;
+  stxAddress: string;
+}
+
+export const buildAssetTransferTx = ({
+  amount,
+  balances,
+  memo,
+  network,
+  nonce,
+  recipient,
+  selectedAsset,
+  senderKey,
+  stxAddress,
+}: AssetTransferTxOptions) => {
+  const { contractAddress, contractName } = selectedAsset;
+  const tokenBalanceKey = Object.keys(balances?.fungible_tokens || {}).find(contract => {
+    return contract.startsWith(contractAddress);
+  });
+  const realAmount = ftUnshiftDecimals(amount, selectedAsset?.meta?.decimals || 0);
+  const postConditionOptions = tokenBalanceKey
+    ? {
+        contractAddress,
+        contractName,
+        assetName: selectedAsset.name,
+        stxAddress,
+        amount: realAmount,
+      }
+    : undefined;
+
+  const postConditions = postConditionOptions ? [makePostCondition(postConditionOptions)] : [];
+  const functionArgs: ClarityValue[] = [
+    uintCV(realAmount),
+    standardPrincipalCVFromAddress(createAddress(stxAddress)),
+    standardPrincipalCVFromAddress(createAddress(recipient)),
+  ];
+  if (selectedAsset.hasMemo) {
+    functionArgs.push(memo !== '' ? someCV(bufferCVFromString(memo)) : noneCV());
+  }
+
+  let txData = {
```

Comment:
`txData` isn't being reassigned, so it can be a const. For me when I see a `let` I assume it's being reassigned elsewhere, otherwise it could be a const. My pref would be to have this rule enabled https://eslint.org/docs/rules/prefer-const. Thoughts?

Reply:
Yes const is better here.